hooks/tk-multi-publish2/maya/publish_lod_geometry.py

```diff
@@ -209,6 +237,83 @@ def publish_files(self, task_settings, item, publish_path):
             instances.
         :param item: Item to process
         :param publish_path: The output path to publish files to
+        """
+        # Creating a temporary file on the publish path, where the alembic from maya would be
+        # exported
+        publish_file_temp = tempfile.NamedTemporaryFile(mode='w+b',
+                                                        suffix='.abc',
+                                                        prefix='tmp',
+                                                        dir="/var/tmp/",
+                                                        delete=True)
+
+        publish_path_temp = publish_file_temp.name.replace("\\", "/")
+
+        # Deciding what type of alembic needs to be exported, based on the current plugin name
+        current_plugin = self.plugin.name
+
+        if current_plugin == "Publish Geometry":
+            # Exporting alembic to a temp location.
+            # This will later be renamed and written to the publish_path
+            self._export_abc_cache(task_settings, item, publish_path_temp)
```

Comment:
The export should happen after we have made sure folder exists. It should be part of the try-except block.

Reply:
There are two "exports" in this situation.
publish_path_temp = "/var/temp/<some_temporary_name>"
publish_path = "/dd/show/..../SHARED/...../<published_file>.abc"

1. Exporting to "publish_path_temp", executed by the method _eex
2. Loading the exported alembic archive at "publish_path_temp" and modifying the top group name, then writing it out to the shotgun "publish_path"

_export_abc_cache() and _export_gpu_abc_cache() export to the "publish_path_temp"
and then _rename_abc_top_group() renames and exports to the publish_path.

Reply:
@the ensuring of the folder's existence, is being done within the "_export_......()" methods for their respective paths.